tests/integration/api/test_aws_integration.py

```diff
@@ -20,25 +19,14 @@
 
 class TestAwsIntegration:
 
-    # List of dictionaries representing the AWS Accounts to cleanup between tests
-    # Ex: [{"account_id": "1234", "role_name": "R1"}, {"account_id": "5678", "role_name": "r2"}]
-    accounts_to_cleanup = []
-
-    @classmethod
-    def setup_class(cls):
-        """ setup any state tied to the execution of the class. called at class
-        level before and after all test methods of the class are called.
-        """
-        initialize(api_key=API_KEY, app_key=APP_KEY, api_host=API_HOST)
-
-        for acc in product(ACCOUNT_IDS, ROLE_NAMES):
-            cls.accounts_to_cleanup.append({'account_id': acc[0], 'role_name': acc[1]})
-
-    def teardown_method(self, method):
-        for account in self.accounts_to_cleanup:
-            dog.AwsIntegration.delete(account_id=account['account_id'], role_name=account.get('role_name'))
+    @pytest.fixture(autouse=True)  # TODO , scope="class"
```

Comment:
```suggestion
    @pytest.fixture(autouse=True)
```

I think it should be executed after each test case so the `TODO` can be removed